isso/ext/notifications.py

```diff
@@ -139,17 +139,43 @@ def format(self, thread, comment, parent_comment, recipient=None, admin=False):
 
         else:
             uri = self.public_endpoint + "/id/%i" % parent_comment["id"]
-            key = self.isso.sign(('unsubscribe', recipient))
+            key = self.isso.sign(('unsubscribe', recipient["email"]))
 
-            rv.write("Unsubscribe from this conversation: %s\n" % (uri + "/unsubscribe/" + quote(recipient) + "/" + key))
+            rv.write("Unsubscribe from this conversation: %s\n" % (uri + "/unsubscribe/" + quote(recipient["email"]) + "/" + key))
 
         rv.seek(0)
         return rv.read()
 
+    def notify_subject(self, thread, comment, parent_comment=None, recipient=None, admin=False):
+        if admin:
+            return self.isso.conf.get("mail", "subject_admin").format(
+                title=thread["title"],
+                replier=comment["author"] or "Anonymous")
+        else:
+            subject_format = list(self.isso.conf.getiter("mail", "subject_user"))
+            if len(subject_format) == 1:
```

Comment:
When would subject_format not have length 1?

Reply:
I use getiter instead of get to get a list of formats. When recipient and parent_comment is the same comment and subject_format has two or more items, then use the second one in the list, otherwise use the first. If subject_format has one item, then the only item will be used in all cases. You might wonder when recipient and parent_comment is different, it's what happens when a subcriber of a reply to the  parent_comment receives the mail about a new reply to the parent_comment.